File: 205. Isomorphic Strings/zip/short.py
class Solution:
    def isIsomorphic(self, s, t):
        if len(s) != len(t):
            return False

        d1, d2 = dict(), dict()
        
        for v, w in zip(s,t):
            if (v in d1 and d1[v] != w) or (w in d2 and d2[w] != v):
                    return False
            d1[v], d2[w] = w, v
        return True
                

File: 205. Isomorphic Strings/zip/short.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if (s.size() != t.size())
            return false;
        unordered_map<char, char> mp;
        unordered_map<char, char> back_mp;

        for (int i = 0; i < s.size(); i++) {
            if ((mp.count(s[i]) && mp[s[i]] != t[i]) || (back_mp.count(t[i]) && back_mp[t[i]] != s[i]))
                return false;
            mp[s[i]] = t[i];
            back_mp[t[i]] = s[i];
        }
        return true;
    }
};